src/toil/__init__.py

```diff
@@ -231,6 +231,22 @@ def customDockerInitCmd():
     return command.replace("'", "'\\''")  # Ensure any single quotes are escaped.
 
 
+def customInitCmd():
+    """
+    Returns the custom command (if any) provided through the ``TOIL_CUSTOM_INIT_COMMAND``
+    environment variable to run prior to running Toil appliance itself in workers and/or the
+    primary node (i.e. this is run one stage before ``TOIL_CUSTOM_DOCKER_INIT_COMMAND``).
+    This can be useful for doing any custom initialization on instances (e.g. authenticating to
+    private docker registries). An empty string is returned if the environment variable is not
+    set.
+    :rtype: str
+    """
+    command = lookupEnvVar(name='user-defined custom init command',
+                           envName='TOIL_CUSTOM_INIT_COMMAND',
+                           defaultValue='')
+    return command.replace("'", "'\\''")  # Ensure any single quotes are escaped.
```

Comment:
This eventually goes into a YAML multiline string where indentation is important. We might want to bail out with an error if the environment variable contains newlines or anything else that could corrupt the YAML we're trying to embed this into that we can't safely escape.

Reply:
That's a good point. Added a sanity check to env var. In practice, I'm not too worried about the edge cases since folks can't use this for malicious purposes (it's their own cluster) and we can always handle them as they come up.